core/templates/dev/head/pages/skill_editor/SkillEditorStateService.ts

```diff
@@ -129,7 +129,7 @@ oppia.factory('SkillEditorStateService', [
         );
       },
 
-      getQuestionSummaries: function(index) {
+      getQuestionSummaries: function(index, fetchMore) {
```

Comment:
Is the added arg ever used? If not please remove it.

/cc @oppia/dev-workflow-team we should be catching this in the linter?

Reply:
It's not used in SkillEditorStateService, but it's used in the same function in TopicEditorStateService, and because in QuestionListDirective it does not specify which one to call (depends on which editor we are currently at), the two functions have to have the same parameters.

Reply:
Hm, that seems a little odd. Why (conceptually) is it used in one and not the other -- is one of the views paged, and the other not?

Reply:
Because in topic_editor page, the number of question_skill_link is inconsistent but it's consistent in skill_editor page (only one skill). That's why I'm using different ways to get question summaries.

Reply:
Sorry, I don't think I understand. It seems to me that, regardless of how many skills we're talking about, it's possible to have more than one "page" of questions. So why wouldn't fetchMore always be a possibility (especially if we are not planning to get all the questions at once, in either case)? And presumably we should name this something like getNextPageOfQuestionSummaries?

Reply:
I think I explained it yesterday. In topic_editor, I fetch 30 links each time, which could be more than I expected, so I keep the unused links in an array, and when going to the next page, only fetch more links when needed. That's the purpose of `fetchMore`. But in skill_editor, there is only one link for each question, so I get 10 links for 10 questions each time. So it has to fetch more every time.

Reply:
But then, if you have this fetcher in a common service, the topic editor page can pass in the necessary true/false value for fetchMore as needed, and the skill editor page can pass in `true` all the time. That would be a cleaner design.